.github/workflows/e2e.yaml

```diff
@@ -83,3 +83,107 @@ jobs:
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v2
+
+      - name: Save the kernel version
+        run: echo "KERNEL_VERSION=$(uname -r)" >> $GITHUB_ENV
+
+      - name: Build the kernel module
+        run: make KERNEL_SRC_DIR="/usr/src/linux-headers-${KERNEL_VERSION}"
+        working-directory: ci/sro-kmod
+
+      - name: Install curl
+        run: |
+          sudo apt -y update
+          sudo apt -y install curl
+
+      - name: Download and install minikube
+        run: |
+          curl -LO https://storage.googleapis.com/minikube/releases/latest/minikube_latest_amd64.deb
+          sudo dpkg -i ./minikube_latest_amd64.deb
+
+      - name: Start minikube and wait until CoreDNS is available
+        run: |
+          minikube start --driver=docker
+          kubectl wait --for=condition=available deployment coredns -n kube-system
+
+      - name: Build sro-kmod
+        run: minikube image build -t sro-kmod:local --build-opt build-arg=KERNEL_VERSION=${KERNEL_VERSION} ci/sro-kmod
+
+      - name: Build SRO
+        run: minikube image build -t sro:local .
+
+      - name: Install kubectl
+        uses: azure/setup-kubectl@v2.0
+
+      - uses: actions/setup-go@v2
+        with:
+          go-version: 1.17.6
+
+      - name: Install NFD
+        run: kubectl apply -k https://github.com/kubernetes-sigs/node-feature-discovery/deployment/overlays/default?ref=v0.10.1
+
+      - name: Deploy SRO
+        run: make deploy
+        env:
+          IMG: sro:local
+
+      - name: Set the SRO manager's pull policy to Never
+        run: kubectl patch deployments.apps -n special-resource-operator special-resource-controller-manager --patch-file ci/patch-sro-manager-pullpolicy.yml
+
+      - name: Setup Helm
+        uses: azure/setup-helm@v1
+
+      - name: Create a ConfigMap containing the test chart
+        run: scripts/make-cm-recipe ci/chart-sro-ci sro-ci
+
+      - name: Check that the sro_ci module is not loaded on the node
+        run: |
+          if minikube ssh -- lsmod | grep sro_ci; then
+            echo "Unexpected lsmod output - the module should not be loaded"
+            exit 1
+          fi
+
+      - name: Create the SpecialResource
+        run: kubectl apply -f ci/sr-sro-ci.yml
+
+      - name: Check that the module gets loaded on the node
+        run: |
+          until minikube ssh -- lsmod | grep sro_ci; do
+            sleep 3
+          done
+        timeout-minutes: 1
+
+      - name: Remove the SpecialResource
+        run: kubectl delete -f ci/sr-sro-ci.yml
+
+      - name: Check that the module gets unloaded from the node
+        run: |
+          until ! minikube ssh -- lsmod | grep sro_ci; do
+            sleep 3
+          done
+        timeout-minutes: 1
+
+      - name: Describe the DaemonSet
+        run: kubectl describe daemonset -n sro-ci sro-ci-driver-container
+        continue-on-error: true # if the job succeeded, there might not be an sro-ci namespace anymore
```

Comment:
I think it's cleaner to specify that the step might fail vs some shell logic. Let me know if you strongly feel otherwise 🙂

Reply:
Maybe I am missing something.

It may fail because the namespace is maybe already deleted after we deleted the CR, therefore, if we want to check the daemonset, why do we do it before we delete the CR and need to depend on some race?

Reply:
These are debug steps at the end of the workflow. They can help you understand what the DaemonSet looked like in case the kernel module was not correctly loaded into and unloaded from the node.
If the namespace does not exist anymore, then the module removal succeeded, then it does not matter if you can't get the troubleshooting data anymore.